halo2_proofs/src/plonk/evaluation.rs

```diff
@@ -534,41 +662,45 @@ impl<C: CurveAffine> Evaluator<C> {
                                 );
 
                                 let r_next = get_rotation_idx(idx, 1, rot_scale, isize);
-                                let r_prev = get_rotation_idx(idx, -1, rot_scale, isize);
 
-                                let a_minus_s =
-                                    permuted_input_coset[idx] - permuted_table_coset[idx];
-                                // l_0(X) * (1 - z(X)) = 0
-                                *value = *value * y + ((one - product_coset[idx]) * l0[idx]);
-                                // l_last(X) * (z(X)^2 - z(X)) = 0
-                                *value = *value * y
-                                    + ((product_coset[idx] * product_coset[idx]
-                                        - product_coset[idx])
-                                        * l_last[idx]);
-                                // (1 - (l_last(X) + l_blind(X))) * (
-                                //   z(\omega X) (a'(X) + \beta) (s'(X) + \gamma)
-                                //   - z(X) (\theta^{m-1} a_0(X) + ... + a_{m-1}(X) + \beta)
-                                //          (\theta^{m-1} s_0(X) + ... + s_{m-1}(X) + \gamma)
-                                // ) = 0
-                                *value = *value * y
-                                    + ((product_coset[r_next]
-                                        * (permuted_input_coset[idx] + beta)
-                                        * (permuted_table_coset[idx] + gamma)
-                                        - product_coset[idx] * table_value)
-                                        * l_active_row[idx]);
-                                // Check that the first values in the permuted input expression and permuted
-                                // fixed expression are the same.
-                                // l_0(X) * (a'(X) - s'(X)) = 0
-                                *value = *value * y + (a_minus_s * l0[idx]);
-                                // Check that each value in the permuted lookup input expression is either
-                                // equal to the value above it, or the value at the same index in the
-                                // permuted table expression.
-                                // (1 - (l_last + l_blind)) * (a′(X) − s′(X))⋅(a′(X) − a′(\omega^{-1} X)) = 0
-                                *value = *value * y
-                                    + (a_minus_s
-                                        * (permuted_input_coset[idx]
-                                            - permuted_input_coset[r_prev])
-                                        * l_active_row[idx]);
+                                let lhs = {
+                                    // τ(X) * Π(φ_i(X)) * (ϕ(gX) - ϕ(X))
+                                    table_value * inputs_prod * (phi_coset[r_next] - phi_coset[idx])
+                                };
+
+                                #[cfg(feature = "logup_skip_inv")]
```

Comment:
"fold(C::Scalar::ZERO, |acc, (_, x)| acc * *x)" looks weird as the result would always be 0. Perhaps, 1 instead of 0 was intended?